application/src/main/java/org/thingsboard/server/service/mail/DefaultMailService.java

```diff
@@ -264,7 +264,11 @@ private void sendMail(TenantId tenantId, CustomerId customerId, TbEmail tbEmail,
                         byte[] bytes = javax.xml.bind.DatatypeConverter.parseBase64Binary(value);
                         String contentType = helper.getFileTypeMap().getContentType(imgId);
                         InputStreamSource iss = () -> new ByteArrayInputStream(bytes);
-                        helper.addInline(imgId, iss, contentType);
+                        if (tbEmail.isAddFileName()){
```

Comment:
Could you please provide some screenshots of "before" and "after" (how does email look)?

Reply:
There is no appearance difference between them.
The difference is only source code of a email
Please check!

![compair_between_with_without_addFileName](https://github.com/thingsboard/thingsboard/assets/53452427/4a51c431-ade8-498e-807c-2a5986d1c35b)

Reply:
For non-HTML email, what do attachments look like? I see that without your feature, the images are named "noname" in the attachments section. What about now?

Reply:
The original behavior of "send email node" does not support Plain text + image attached email.
If "msg.html=true" does not set, HTML text is sent as Plain text and there is no attachment file.

So I modified my code to support Plain text and attached file email.

I made a screen capture to compare between

1. HTML without attach filename (html=true/addFileName=false)
2. HTML with attach filename (html=true/addFileName=true)
3. Plain text with attach filename (html=false/addFileName=true/false)
    -> Cannot make Plain text without attach filename by addAttachment function.
    addAttachment  function add "name" to Content-Type automatically.

Please check it.

![PR11132_compair_html_plain_text](https://github.com/thingsboard/thingsboard/assets/53452427/4e18a9bd-9572-4511-a6c1-70d76598dfb4)